src/hooks/useComparison.ts

```diff
@@ -0,0 +1,47 @@
+import { useDispatch } from 'react-redux';
+
+import { RootState } from '../common/store';
+import { comparisonResults as secondRevisionResults } from '../mockData/9d5066525489';
+import { comparisonResults as thirdRevisionResults } from '../mockData/a998c42399a8';
+import { comparisonResults as firstRevisionResults } from '../mockData/bb6a5e451dac';
+import { setCompareData } from '../reducers/CompareResults';
+import { updateComparison } from '../reducers/ComparisonSlice';
+import { Strings } from '../resources/Strings';
+import { truncateHash } from '../utils/helpers';
+import { useAppSelector } from './app';
+
+const useComparison = () => {
+  const dispatch = useDispatch();
+  const comparisonResults = firstRevisionResults.concat(
+    secondRevisionResults,
+    thirdRevisionResults,
+  );
+  const activeComparison: string = useAppSelector(
+    (state: RootState) => state.comparison.activeComparison,
+  );
+  const allRevisions =
+    Strings.components.comparisonRevisionDropdown.allRevisions;
+
+  const switchComparisonData = () => {
+    if (activeComparison === truncateHash(firstRevisionResults[0].new_rev)) {
+      dispatch(setCompareData({ data: firstRevisionResults }));
+    }
+    if (activeComparison === truncateHash(secondRevisionResults[0].new_rev)) {
+      dispatch(setCompareData({ data: secondRevisionResults }));
+    }
+    if (activeComparison === truncateHash(thirdRevisionResults[0].new_rev)) {
+      dispatch(setCompareData({ data: thirdRevisionResults }));
+    }
+    if (activeComparison === allRevisions) {
+      dispatch(setCompareData({ data: comparisonResults }));
+    }
+  };
+
+  const handlerChangeComparison = (selectedRevision: string) => {
+    dispatch(updateComparison({ activeComparison: selectedRevision }));
```

Comment:
I have several comments for this piece of code.

### 1. Keeping the same store structure, I believe we could call `switchComparisonData` directly here instead of using an effect in ResultsView.

Instead you're doing this dance:
1. The user changes the option.
2. We dispatch this action to change the string.
3. React renders the UI.
4. `useEffect` is called in the other component. This component calls `switchComparisonData`.
5. `switchComparisonData` dispatches again.
6. React renders the UI again.

This isn't performant, and this introduces inter-dependencies between the hook and the user of the hook, while the hook should better be more self-contained. With my proposal it would be much more self-contained.

See https://react.dev/learn/you-might-not-need-an-effect#chains-of-computations for a more complete explanation.

### 2. But another structure is possible

Instead of storing the full comparison data in the store, it could be _computed_ in a selector.

The store only has the source data to compute the comparison data:
* the full data (with `setCompareData`)
* the revision value

Then the store will _compute_ the returned value from the full data and the revision value. To handle the fakeData params, we can put the fake data as the full data during the fetch operation.

I think this is both easier to code, easier to follow, and easier to maintain.
Also I believe that the current code as it is now works fine with the fake data but will need to be redone with the full data. With the approach I'm suggesting here, the code would work with the fake data and with the real data.

Tell me what you think!